helix-lsp/src/client.rs

```diff
@@ -1114,6 +1115,26 @@ impl Client {
         Some(self.call::<lsp::request::InlayHintRequest>(params))
     }
 
+    pub fn text_document_colors(
+        &self,
+        text_document: lsp::TextDocumentIdentifier,
+        work_done_token: Option<lsp::ProgressToken>,
+    ) -> Option<impl Future<Output = Result<Option<Vec<lsp::ColorInformation>>>>> {
+        self.capabilities.get().unwrap().color_provider.as_ref()?;
+        let params = lsp::DocumentColorParams {
+            text_document,
+            work_done_progress_params: lsp::WorkDoneProgressParams {
+                work_done_token: work_done_token.clone(),
+            },
+            partial_result_params: helix_lsp_types::PartialResultParams {
+                partial_result_token: work_done_token,
+            },
+        };
+
+        let res = self.call::<lsp::request::DocumentColor>(params);
+        Some(async move { Ok(serde_json::from_value(res.await?)?) })
```

Comment:
See https://github.com/helix-editor/helix/commit/7e7a98560e58371690d00e380d79e0bbaaf19d54, this can now become:

```suggestion
        Some(self.call::<lsp::request::DocumentColor>(params))
```

It looks like the return type will also need to be updated to drop the inner option as the response is just the vec: https://github.com/helix-editor/helix/blob/f07c1c1b29add205841877d6665277c007fff83e/helix-lsp-types/src/request.rs#L627